### PR TITLE
Changed  the input recorder to store actions by strings

### PR DESCRIPTION
**Pull request by druiz17**
Made the input recorder store recorded actions by string and not int.

### CODE REVIEW

**Comment by druiz17** [comment]
Test plan
- run the script inputRecorder.js 
- click start recording, record some movement.
- click stop recording and then save the recording
- click load and load the file you recently saved. playback that recording
- check if the correct recording is played back.

- download the file below.
- check if the file loads and plays.

[27_Matthew_Side-Splits.gz](https://github.com/highfidelity/hifi/files/1045351/27_Matthew_Side-Splits.gz)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/301/HighFidelity-Beta-PR10588-cca0fa66004464713641aef86bd73a75ba4963bd.exe

**Comment by druiz17** [comment]
test this please

**Comment by CainFoool** [comment]
Testing this.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/303/HighFidelity-Beta-PR10588-cca0fa66004464713641aef86bd73a75ba4963bd.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/308/HighFidelity-Beta-PR10588-c686acc071002c14fe0410c8a0ca778cd8276824.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/308/HighFidelity-Beta-PR10588-c686acc071002c14fe0410c8a0ca778cd8276824.dmg

**Comment by CainFoool** [comment]
@druiz17 Will I test this with the final change commit?

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/309/HighFidelity-Beta-PR10588-0e82b33d6d4cf9cadccde4abd9b33a127d0003b3.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/309/HighFidelity-Beta-PR10588-0e82b33d6d4cf9cadccde4abd9b33a127d0003b3.dmg

**Comment by CainFoool** [comment]
Testing.

**Comment by druiz17** [comment]
This should be the final commit change

**Comment by CainFoool** [comment]
# Test Passed

Recording worked as expected, recording from OP played properly.

File: libraries/controllers/src/controllers/InputRecorder.cpp [hunk 1]
```diff
@@ -181,48 +198,77 @@ namespace controller {
...
+        if (success && keyValue != data.end()) {
             _framesRecorded = data["frameCount"].toInt();
             QJsonArray actionArrayList = data["actionList"].toArray();
             QJsonArray poseArrayList = data["poseList"].toArray();
 
             for (int actionIndex = 0; actionIndex < actionArrayList.size(); actionIndex++) {
                 QJsonArray actionState = actionArrayList[actionIndex].toArray();
                 for (int index = 0; index < actionState.size(); index++) {
-                    _currentFrameActions[index] = actionState[index].toDouble();
+                    QJsonObject actionObject = actionState[index].toObject();;
```
Comment (sethalves) [inline libraries/controllers/src/controllers/InputRecorder.cpp]: double ;

Comment (druiz17) [inline libraries/controllers/src/controllers/InputRecorder.cpp]: Each index is an object that holds the name of the action and value. see the next line

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/320/HighFidelity-Beta-PR10588-725388043f2cd1d7ebbf82f977a8fdfd82e2c331.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/320/HighFidelity-Beta-PR10588-725388043f2cd1d7ebbf82f977a8fdfd82e2c331.dmg

**Pull request merged**